annotation/annotation/tasks/services.py

```diff
@@ -62,6 +62,26 @@
     ValidationSchema,
 )
 
+
+class LRU(OrderedDict):
+    def __init__(self, capacity):
+        super().__init__()
+        self.capacity = capacity
+
+    def __getitem__(self, key):
+        value = super().__getitem__(key)
+        self.move_to_end(key)
+        return value
+
+    def __setitem__(self, key, value):
+        if key in self:
+            self.move_to_end(key)
+        super().__setitem__(key, value)
```

Comment:
what if we reorder them? first - set and then move to end. this way we dont have to check if key exists